core/src/test/java/io/grpc/internal/DelayedStreamTest.java

```diff
@@ -0,0 +1,172 @@
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.internal;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.google.common.util.concurrent.MoreExecutors;
+
+import io.grpc.Codec;
+import io.grpc.DecompressorRegistry;
+import io.grpc.IntegerMarshaller;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.MethodDescriptor.MethodType;
+import io.grpc.Status;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.concurrent.Executor;
+
+/**
+ * Tests for {@link DelayedStream}.  Most of the state checking is enforced by
+ * {@link ClientCallImpl} so we don't check it here.
+ */
+@RunWith(JUnit4.class)
+public class DelayedStreamTest {
+  private static final Executor executor = MoreExecutors.directExecutor();
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Mock private ClientStreamListener listener;
+  @Mock private ClientTransport transport;
+  @Mock private ClientStream realStream;
+  @Captor private ArgumentCaptor<Status> statusCaptor = ArgumentCaptor.forClass(Status.class);
+  private DelayedStream stream;
+  private Metadata headers = new Metadata();
+
+  private MethodDescriptor<Integer, Integer> method = MethodDescriptor.create(
+      MethodType.UNARY, "service/method", new IntegerMarshaller(), new IntegerMarshaller());
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    stream = new DelayedStream(listener, executor);
+  }
+
+  @Test
+  public void realStreamCreatedWhenTransportReady() {
+    stream.setStream(realStream);
+
+    verify(transport).newStream(method, headers, listener);
+  }
+
+  @Test
+  public void createStream_sendsAllMessages() {
```

Comment:
I have updated the test names in DelayedStreamTest, and added some simple
tests in CCI.  PTAL

On Tue, Dec 1, 2015 at 11:37 AM, Kun Zhang <notifications@github.com> wrote:

> In core/src/test/java/io/grpc/internal/DelayedStreamTest.java
> <https://github.com/grpc/grpc-java/pull/1234#discussion_r46327066>:
>
> > +  @Before
> > +  public void setUp() {
> > +    MockitoAnnotations.initMocks(this);
> > +
> > +    stream = new DelayedStream(listener, executor);
> > +  }
> > +
> > +  @Test
> > +  public void realStreamCreatedWhenTransportReady() {
> > +    stream.setStream(realStream);
> > +
> > +    verify(transport).newStream(method, headers, listener);
> > +  }
> > +
> > +  @Test
> > +  public void createStream_sendsAllMessages() {
>
> These should be renamed to setStream_* too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-java/pull/1234/files#r46327066>.
>